setup.py

```diff
@@ -318,20 +318,20 @@ def run(self):
     'pandas',
 ]
 
-NUMPY_STR = 'numpy >= 1.17.0'
+NUMPY_STR = 'numpy >= 1.23.4'
 #
 # We pin the Cython version for reproducibility.  We expect our extensions
 # to build with any sane version of Cython, so we should update this pin
 # periodically.
 #
-CYTHON_STR = 'Cython==0.29.28'
+CYTHON_STR = 'Cython==0.29.32'
 
 # Allow overriding the Cython version requirement
 CYTHON_STR = os.environ.get('GENSIM_CYTHON_REQUIRES', CYTHON_STR)
 
 install_requires = [
     NUMPY_STR,
-    'scipy >= 0.18.1',
+    'scipy >= 1.9.3',
```

Comment:
This looks like the very latest version of scipy.

We want the *oldest* version (that works) here, because otherwise we make life difficult for users who don't have the very latest.

Could you please find which is the first version of numpy & scipy to support 3.11?

Reply:
> Could you please find which is the first version of numpy & scipy to support 3.11?

just check(install on local..pip)
the last version of scipy and numpy are 1.9.2 and 1.23.0 (not very far from the lastest)

maybe we can add special condition just for python 3.11 ?
something like overriding version(numpy,scipy) like the cython https://github.com/RaRe-Technologies/gensim/blob/c93eb0b647ec9de7cfcb43c74c37f295f2944821/setup.py#L330

Reply:
According to https://devguide.python.org/versions/, the currently oldest supported Python (before end-of-life) is Python 3.7. So we can ignore Pythons older than that (3.6 etc).

Now looking at SciPy's compatibility matrix https://docs.scipy.org/doc/scipy/dev/toolchain.html :

<img width="785" alt="Screen Shot 2022-11-08 at 10 22 27" src="https://user-images.githubusercontent.com/610412/200526037-dee385ec-7d32-4f86-8a59-9218ec3e19cc.png">

It seems for 3.11 we need Scipy 1.9.0 (or .1?) and Numpy 1.18.5. Which doesn't match your `the oldest version that support 3.11 for scipy and numpy are 1.9.2 and 1.23.0 ` – can you verify please?

> maybe we can add special condition just for python 3.11 ?

I don't think so. Gensim is a library used by ML developers, so we can afford to update its dependencies. We just shouldn't do it without a reason. @mpenkov WDYT?

Reply:
Yes, I think that's fine. The versions you suggested also make sense to me.

Reply:
ah thank you for detail and correction

my missunderstanding
i refer to this issue https://github.com/scipy/scipy/issues/16851#issuecomment-1271363490 for oldest "wheel" scipy 1.9.2 and https://numpy.org/news/ for numpy


> If the matrix above is correct, I'd go with Scipy>=1.9.0 and Numpy>=1.18.5.

just tried with this version and  it works

Reply:
@piskvorky @acul3 Does this mean that 3.6 and 3.7 are no longer supported? If so, we need to update setup.py accordingly.

Reply:
> the currently oldest supported Python (before end-of-life) is Python 3.7. So we can ignore Pythons older than that (3.6 etc).

hmm it seems python 3.7 still need supported according to this argument

Reply:
From the table above, it looks like it's impossible to support both 3.7 and 3.11 using a single version of scipy, numpy, etc.

So, our options are:

1. Use different versions for Py3.11
2. Drop 3.7 support now (3.7 reaches EOL in 6 months anyway)
3. Wait before 3.7 reaches EOL, and then add support for 3.11

Personally, I think 2) is best. 1) requires a fair bit of effort for very little value (6 more months of 3.7 support, and after 6 months the value of that effort goes down to zero). We can do a major version bump. If people really need support for 3.7, they can continue to use the current version of gensim.

@piskvorky thoughts?

Reply:
Yes, agreed. Now that Python versions go away frequently, there isn't much expectation to support older ones (not as much as when py2.7 was supported for ages).